### ISSUE TITLE
Run 2021-04 ConMon scans

### ISSUE THREAD

**Issue by pburkholder** [cloud-gov/product#1556]
In order for us to update the JAB on our compliance in a consistent way, we need to run Continuous Monitoring scans on approximately the 23rd of the month. (If this date falls on a weekend or federal holiday, adjust to the last business day before the date.)

For context, see our [Continuous Monitoring Strategy](https://cloud.gov/docs/ops/continuous-monitoring/), including [the monthly reporting summary explanation](https://cloud.gov/docs/ops/continuous-monitoring/#monthly-reporting-summary). 

## Netsparker:

We are also uing GSA's Netsparker scanning service. Four days before our
scheduled scan, [submit a Security Scan
Request](https://gsa.servicenowservices.com/sp?id=sc_cat_item&sys_id=611282de6fa13100b544204fae3ee40c)
on GSA Service now. Complete as follows:

* Planning Date: (23rd or nearest working day to that)
* System POC: select the cloud.gov compliance lead
* Technical POC: select yourself 
* Service/Staff office: FAS (Q)
* GSA FISMA system: Federalist
* Site/System Descriptive name: cloud.gov
* Site/System URL: (see additional comments)
* Site/System Phasae: Production
* Type of Scan: ATO or Certification Scan
* Does the site require authentication: Yes
* Username of the account for scanning: The scan operator should use their own cloud.gov account through GSA SecureAuth
* Explanation: Monthly Conmon for FedRAMP 
* Justification: Monthly Conmon for FedRAMP 

Additional Comments:

> Please provide PDF and XML results for scans of the cloud.gov sites:
> ```
> cloud.gov
> account.fr.cloud.gov
> dashboard.fr.cloud.gov
> idp.fr.cloud.gov
> login.fr.cloud.gov
> logs.fr.cloud.gov

> The scan operator will need a cloud.gov account. For that, browse to https://dashboard.fr.cloud.gov, select "Agree and Continue", choose the GSA identity provider, and sign in with GSA SecureAuth. Once that's complete the authenticated site scans should work.

> Thanks much -- cloud.gov team

```
```

## Disk Usage

A single ZAP scan of the cloud-gov context requires significant disk space (over 100GB). If you have run ZAP previously, you should check to see if you previous sessions have been persisted. If so, you likely need to clear out those directories before proceeding.

You can check ZAP's disk usage with:

```
du -h -d 1 ~/Library/Application\ Support/ZAP/
```

If you see an abnormally large `session` or `sessions` directory (my last run was 132G), you likely want to delete all files in these directories before proceeding.

## Running ZAP Scans

ZAP scans take hours. We recommend you start in the morning. There are two separate scans to run and the second one takes considerably longer than the first.

- Check that you have the [latest stable version of ZAP](https://www.zaproxy.org/download/) or install via Homebrew with `brew cask install (or brew upgrade --cask) owasp-zap` (recommended). Upgrade if necessary.

     > NOTE: If you see an error running ZAP as an unsigned application, run the following from the command line: `xattr -dr com.apple.quarantine '/Applications/OWASP ZAP.app'`

- Be sure you have Firefox installed (with Homebrew `brew cask install firefox` or any way you chose). Chrome does not support proxy settings while Firefox does.

- Configure ZAP as a proxy for Firefox: https://www.zaproxy.org/docs/desktop/start/proxies/

- Download the [cloud.gov conmon](https://raw.githubusercontent.com/18F/cg-product/master/cloud.gov-conmon.context) cloud.gov context from this repository and update it if necessary

- Load the cloud.gov context into ZAP (File > Import Context)
     
- You have to log in to each site in the context list so that ZAP can scan inside the authenticated websites. You have to be on the VPN. In Firefox, log in to each site in the context list. You can see the list by clicking the `cloud.gov common` context > Include in Context.

- To prevent getting noise in the scan results (since that causes major confusion when the FedRAMP team processes the ConMon report), review the `Sites` list to ensure only the cloud.gov sites have a small red circle/sight on them (denoting the site will be included). Remove any sites not needed by CTRL-clicking on them and selecting `Delete`.

- CTRL-click on the `cloud.gov common` context and run the `Spider` scan.  This takes a little less than an hour.

- After the `Spider` scan is complete, again CTRL-click on the `cloud.gov common` context and this time run the `Active Scan`. 

- After both scans are complete, export the results as both XML and HTML from the `Reports` menu. Name the files according to `YYYYMMDD-ZAP.xml/html`.

- [x] Upload both reports to Google Drive: https://drive.google.com/drive/u/0/folders/0B5fn0WMJaYDnaFdCak5WNWRGb1U in a folder named `YYYYMMDD-ZAP-Nessus`.

You can shut down ZAP and Firefox. If you use Firefox in your day to day, you likely want to remove the proxy.

## Export Nessus Scans

- Log in to Nessus: https://nessus.fr.cloud.gov/

- Select `All Scans`

- Click on each scan except for staging, and export the .nessus file (Export > Nessus) and the Executive Summary report (Report > HTML).

- [x] Upload all exports (both Executive Summary and .nessus files) to the same Google Drive folder you put the ZAP scans in.

## Update the POAM Inventory sheet

A python script is used to generate the inventory list.

- Open the [POAM Inventory sheet](https://docs.google.com/spreadsheets/d/1_9Neq8fGO4NdQhsqLXDn445g3GUa1k_FZUrUXc7hulY/edit#gid=1371600163)

- Delete the data rows (**starting after the RDS database listings**) - These rows are locked to prevent inadvertent editing.

- For the tooling and production jumpboxes, login to each of them and from the initial home directory you start in, run `python3 cg-scripts/generate-POAM-inventory.py`. This will output data to your terminal window in CSV format. Copy the entire CSV output.

- Paste the contents in the spreadsheet by selecting the first cell in row 3 then pasting with CTRL-Shift-V to paste without formatting. Then select the paste icon that appears and click `Split text to columns`

- [x] Verify you have pasted the inventory for both production and tooling.

**Comment by pburkholder** [comment]
Netsparker: Request Number : REQ0602458

**Comment by ccostino** [comment]
I'll get the OWASP scans done this morning (4/23/21).

**Comment by ccostino** [comment]
The VPN is being flaky today, we're going to try the OWASP ZAP scans again on Monday, 4/26/2021.

**Comment by ccostino** [comment]
Nessus scan results have been uploaded and the POA&M Inventory spreadsheet has been updated.

**Comment by ccostino** [comment]
I attempted to run the OWASP ZAP scans again yesterday (4/26/2021); they went for over 14 hours, the scan still wasn't finished, and I had been knocked off the VPN after 12 hours.  The ZAP app also became entirely unresponsive and seemed to pull in a bunch of other sites to scan even after I configured it not to do that and focus only on the cloud.gov sites defined in our context.

I ended up having to force quit the app and reboot my machine to get things working again.  I need someone else to run these scans, I'm not sure why this didn't work right for me.
